### ISSUE TITLE
[Triton] Add sdpa pattern support for `BertForQuestionAnswering` model

### ISSUE THREAD

**Issue by hoshibara** [pytorch/pytorch#126057]
### 🚀 The feature, motivation and pitch

In a `BertForQuestionAnswering` model, the core calculation process of the `BertSelfAttention.forward` function can be simplified as:
```python
key_layer = self.transpose_for_scores(self.key(hidden_states))
value_layer = self.transpose_for_scores(self.value(hidden_states))
query_layer = self.transpose_for_scores(mixed_query_layer)
attention_scores = torch.matmul(query_layer, key_layer.transpose(-1, -2))
attention_scores = attention_scores / math.sqrt(self.attention_head_size)
if attention_mask is not None:
    attention_scores = attention_scores + attention_mask
attention_probs = nn.functional.softmax(attention_scores, dim=-1)
attention_probs = self.dropout(attention_probs)
context_layer = torch.matmul(attention_probs, value_layer)
```
During inference, `dropout` layers will be treated as a clone layers. But the graph cannot match the existing pattern.
I hope `torch.compile` can support this static graph.

cc @ezyang @msaroufim @bdhirsh @anijain2305 @chauhang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire

**Comment by xmfan** [comment]
cc @Chillee something that could be solved with templated attention?

**Comment by eellison** [comment]
Are you sure this doesn't already work ? 

```
TORCH_LOGS="output_code" python benchmarks/dynamo/huggingface.py --performance --inference --bfloat16 --backend inductor --device cuda --only BertForQuestionAnswering
```

shows 12 `_scaled_dot_product_flash_attention` invocations

**Comment by eellison** [comment]
closing, reopen if you think it should be open.

**Comment by hoshibara** [comment]
Here is my test script.
In this script, the existing patterns cannot match the graph in `BertSelfAttention`.

```python
import torch
from transformers import AutoModelForQuestionAnswering, BertTokenizer

bert_qa_model = AutoModelForQuestionAnswering.from_pretrained('./bert-base-uncased').to('cuda')
bert_tokenizer = BertTokenizer.from_pretrained('./bert-base-uncased')

question = "What are the main attractions in Paris?"
context = (
    "Paris, the capital of France, is known for its rich history, art, and culture. Some of the main attractions in Paris include "
    "the Eiffel Tower, the Louvre Museum, Notre-Dame Cathedral, and the Champs-Élysées. The city is also famous for its charming "
    "neighborhoods, such as Montmartre and Le Marais, as well as its beautiful parks and gardens like the Luxembourg Gardens and "
    "Tuileries Garden. Visitors to Paris can enjoy world-class dining, shopping, and entertainment, making it a top destination for "
    "tourists from around the world."
    "Paris, the capital of France, is known for its rich history, art, and culture. Some of the main attractions in Paris include "
    "the Eiffel Tower, the Louvre Museum, Notre-Dame Cathedral, and the Champs-Élysées. The city is also famous for its charming "
    "neighborhoods, such as Montmartre and Le Marais, as well as its beautiful parks and gardens like the Luxembourg Gardens and "
    "Tuileries Garden. Visitors to Paris can enjoy world-class dining, shopping, and entertainment, making it a top destination for "
    "tourists from around the world."
    "Paris, the capital of France, is known for its rich history, art, and culture. Some of the main attractions in Paris include "
    "the Eiffel Tower, the Louvre Museum, Notre-Dame Cathedral, and the Champs-Élysées. The city is also famous for its charming "
    "neighborhoods, such as Montmartre and Le Marais, as well as its beautiful parks and gardens like the Luxembourg Gardens and "
    "Tuileries Garden. Visitors to Paris can enjoy world-class dining, shopping, and entertainment, making it a top destination for "
    "tourists from around the world."
)
inputs = bert_tokenizer.encode_plus(question, context, return_tensors="pt")
input_ids = inputs["input_ids"].to('cuda')
input_ids = input_ids.expand(32, -1)
attention_mask = inputs["attention_mask"].to('cuda')
attention_mask = attention_mask.expand(32, -1)

bert_qa_model.eval()
bert_qa_model.to(torch.bfloat16)
bert_qa_model = torch.compile(bert_qa_model)

output = bert_qa_model(input_ids=input_ids, attention_mask=attention_mask)

with torch.profiler.profile() as prof:
    with torch.inference_mode():
        for i in range(10):
            output = bert_qa_model(input_ids=input_ids, attention_mask=attention_mask)
            prof.step()
            
print(prof.key_averages().table(sort_by="self_cuda_time_total"))
```
